spring-data-couchbase/src/main/java/org/springframework/data/couchbase/core/TransactionalSupport.java

```diff
@@ -50,6 +52,14 @@ public static Mono<Optional<CouchbaseResourceHolder>> checkForTransactionInThrea
 		});
 	}
 
+	public static Optional<CouchbaseResourceHolder> checkForTransactionInThreadLocalStorage(ContextView ctx) {
+		return Optional.ofNullable(ctx.hasKey(TransactionMarker.class) ? new CouchbaseResourceHolder(ctx.get(TransactionMarker.class).context()) : null);
```

Comment:
This isn't too great as a) it's not DRY or encapsulated - this functionality would really belong in TransactionMarkerOwner, and b) most importantly, it's removing a critical check.  We won't know if we're inside a blocking transaction here, as we don't have the ThreadLocal check.

Reply:
> We won't know if we're inside a blocking transaction here, as we don't have the ThreadLocal check.
Both checks are there, however if the ctx is not due to the transaction, and there is indeed a blocking (ThreadLocal) transaction, we will miss it.  Hence the request for a TransactionMarkerOwner api that does require a ctx and return a Mono to get the ThreadLocal core.
```
	boolean isExistingTransaction = ctx != null ? TransactionalSupport.checkForTransactionInThreadLocalStorage(ctx).isPresent() :
				TransactionalSupport.checkForTransactionInThreadLocalStorage().block().isPresent();
```


Right.  So (a) requires the modification to TransactionMarkerOwner, and (b) needs what I've described in the other comment.